h2/src/main/org/h2/command/Parser.java

```diff
@@ -1855,6 +1860,34 @@ private String readFromAlias(String alias) {
         return null;
     }
 
+    private void discardWithTableHints() {
+        if (readIf("WITH")) {
+            read(OPEN_PAREN);
+            if (!readIf(CLOSE_PAREN)) {
+                do {
+                    discardTableHint();
+                } while (readIfMore(true));
+            }
+        }
+    }
+
+    private void discardTableHint() {
+        if (readIf("INDEX")) {
+            if (readIf(OPEN_PAREN)) {
+                do {
+                    readExpression();
```

Comment:
Is `index_value` really an expression? What exactly is allowed here in SQL Server? I cannot find the answer in documentation of SQL Server.

Reply:
I honestly don't know - any suggestions for a `readExpression` replacement?

Reply:
If `readExpression()` works fine with real-world use cases just leave it as is for now.